File: src/java/SQ_Energy_Balance/EnergybalanceAuxiliary.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class EnergyBalanceAuxiliary
{
    private Double minTair;
    private Double maxTair;
    private Double solarRadiation;
    private Double vaporPressure;
    private Double extraSolarRadiation;
    private Double hslope;
    private Double plantHeight;
    private Double wind;
    private Double deficitOnTopLayers;
    private Double VPDair;
    private Double netRadiation;
    private Double netOutGoingLongWaveRadiation;
    private Double netRadiationEquivalentEvaporation;
    private Double energyLimitedEvaporation;
    private Double soilEvaporation;
    
    public EnergyBalanceAuxiliary() { }
    
    public EnergyBalanceAuxiliary(EnergyBalanceAuxiliary toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.minTair = toCopy.getminTair();
            this.maxTair = toCopy.getmaxTair();
            this.solarRadiation = toCopy.getsolarRadiation();
            this.vaporPressure = toCopy.getvaporPressure();
            this.extraSolarRadiation = toCopy.getextraSolarRadiation();
            this.hslope = toCopy.gethslope();
            this.plantHeight = toCopy.getplantHeight();
            this.wind = toCopy.getwind();
            this.deficitOnTopLayers = toCopy.getdeficitOnTopLayers();
            this.VPDair = toCopy.getVPDair();
            this.netRadiation = toCopy.getnetRadiation();
            this.netOutGoingLongWaveRadiation = toCopy.getnetOutGoingLongWaveRadiation();
            this.netRadiationEquivalentEvaporation = toCopy.getnetRadiationEquivalentEvaporation();
            this.energyLimitedEvaporation = toCopy.getenergyLimitedEvaporation();
            this.soilEvaporation = toCopy.getsoilEvaporation();
            this.netRadiation = toCopy.getnetRadiation();
            this.netRadiationEquivalentEvaporation = toCopy.getnetRadiationEquivalentEvaporation();
            this.energyLimitedEvaporation = toCopy.getenergyLimitedEvaporation();
            this.soilEvaporation = toCopy.getsoilEvaporation();
        }
    }
    public Double getminTair()
    { return minTair; }

    public void setminTair(Double _minTair)
    { this.minTair= _minTair; } 
    
    public Double getmaxTair()
    { return maxTair; }

    public void setmaxTair(Double _maxTair)
    { this.maxTair= _maxTair; } 
    
    public Double getsolarRadiation()
    { return solarRadiation; }

    public void setsolarRadiation(Double _solarRadiation)
    { this.solarRadiation= _solarRadiation; } 
    
    public Double getvaporPressure()
    { return vaporPressure; }

    public void setvaporPressure(Double _vaporPressure)
    { this.vaporPressure= _vaporPressure; } 
    
    public Double getextraSolarRadiation()
    { return extraSolarRadiation; }

    public void setextraSolarRadiation(Double _extraSolarRadiation)
    { this.extraSolarRadiation= _extraSolarRadiation; } 
    
    public Double gethslope()
    { return hslope; }

    public void sethslope(Double _hslope)
    { this.hslope= _hslope; } 
    
    public Double getplantHeight()
    { return plantHeight; }

    public void setplantHeight(Double _plantHeight)
    { this.plantHeight= _plantHeight; } 
    
    public Double getwind()
    { return wind; }

    public void setwind(Double _wind)
    { this.wind= _wind; } 
    
    public Double getdeficitOnTopLayers()
    { return deficitOnTopLayers; }

    public void setdeficitOnTopLayers(Double _deficitOnTopLayers)
    { this.deficitOnTopLayers= _deficitOnTopLayers; } 
    
    public Double getVPDair()
    { return VPDair; }

    public void setVPDair(Double _VPDair)
    { this.VPDair= _VPDair; } 
    
    public Double getnetRadiation()
    { return netRadiation; }

    public void setnetRadiation(Double _netRadiation)
    { this.netRadiation= _netRadiation; } 
    
    public Double getnetOutGoingLongWaveRadiation()
    { return netOutGoingLongWaveRadiation; }

    public void setnetOutGoingLongWaveRadiation(Double _netOutGoingLongWaveRadiation)
    { this.netOutGoingLongWaveRadiation= _netOutGoingLongWaveRadiation; } 
    
    public Double getnetRadiationEquivalentEvaporation()
    { return netRadiationEquivalentEvaporation; }

    public void setnetRadiationEquivalentEvaporation(Double _netRadiationEquivalentEvaporation)
    { this.netRadiationEquivalentEvaporation= _netRadiationEquivalentEvaporation; } 
    
    public Double getenergyLimitedEvaporation()
    { return energyLimitedEvaporation; }

    public void setenergyLimitedEvaporation(Double _energyLimitedEvaporation)
    { this.energyLimitedEvaporation= _energyLimitedEvaporation; } 
    
    public Double getsoilEvaporation()
    { return soilEvaporation; }

    public void setsoilEvaporation(Double _soilEvaporation)
    { this.soilEvaporation= _soilEvaporation; } 
    
}

File: src/cs/SQ_Energy_Balance/EnergyBalanceAuxiliary.cs
using System;
using System.Collections.Generic;

public class EnergyBalanceAuxiliary 
{
    private double _minTair;
    private double _maxTair;
    private double _solarRadiation;
    private double _vaporPressure;
    private double _extraSolarRadiation;
    private double _hslope;
    private double _plantHeight;
    private double _wind;
    private double _deficitOnTopLayers;
    private double _VPDair;
    private double _netRadiation;
    private double _netOutGoingLongWaveRadiation;
    private double _netRadiationEquivalentEvaporation;
    private double _energyLimitedEvaporation;
    private double _soilEvaporation;
    
        public EnergyBalanceAuxiliary() { }
    
    
    public EnergyBalanceAuxiliary(EnergyBalanceAuxiliary toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    minTair = toCopy.minTair;
    maxTair = toCopy.maxTair;
    solarRadiation = toCopy.solarRadiation;
    vaporPressure = toCopy.vaporPressure;
    extraSolarRadiation = toCopy.extraSolarRadiation;
    hslope = toCopy.hslope;
    plantHeight = toCopy.plantHeight;
    wind = toCopy.wind;
    deficitOnTopLayers = toCopy.deficitOnTopLayers;
    VPDair = toCopy.VPDair;
    netRadiation = toCopy.netRadiation;
    netOutGoingLongWaveRadiation = toCopy.netOutGoingLongWaveRadiation;
    netRadiationEquivalentEvaporation = toCopy.netRadiationEquivalentEvaporation;
    energyLimitedEvaporation = toCopy.energyLimitedEvaporation;
    soilEvaporation = toCopy.soilEvaporation;
    }
    }
    public double minTair
        {
            get { return this._minTair; }
            set { this._minTair= value; } 
        }
    public double maxTair
        {
            get { return this._maxTair; }
            set { this._maxTair= value; } 
        }
    public double solarRadiation
        {
            get { return this._solarRadiation; }
            set { this._solarRadiation= value; } 
        }
    public double vaporPressure
        {
            get { return this._vaporPressure; }
            set { this._vaporPressure= value; } 
        }
    public double extraSolarRadiation
        {
            get { return this._extraSolarRadiation; }
            set { this._extraSolarRadiation= value; } 
        }
    public double hslope
        {
            get { return this._hslope; }
            set { this._hslope= value; } 
        }
    public double plantHeight
        {
            get { return this._plantHeight; }
            set { this._plantHeight= value; } 
        }
    public double wind
        {
            get { return this._wind; }
            set { this._wind= value; } 
        }
    public double deficitOnTopLayers
        {
            get { return this._deficitOnTopLayers; }
            set { this._deficitOnTopLayers= value; } 
        }
    public double VPDair
        {
            get { return this._VPDair; }
            set { this._VPDair= value; } 
        }
    public double netRadiation
        {
            get { return this._netRadiation; }
            set { this._netRadiation= value; } 
        }
    public double netOutGoingLongWaveRadiation
        {
            get { return this._netOutGoingLongWaveRadiation; }
            set { this._netOutGoingLongWaveRadiation= value; } 
        }
    public double netRadiationEquivalentEvaporation
        {
            get { return this._netRadiationEquivalentEvaporation; }
            set { this._netRadiationEquivalentEvaporation= value; } 
        }
    public double energyLimitedEvaporation
        {
            get { return this._energyLimitedEvaporation; }
            set { this._energyLimitedEvaporation= value; } 
        }
    public double soilEvaporation
        {
            get { return this._soilEvaporation; }
            set { this._soilEvaporation= value; } 
        }
}